lms/djangoapps/lti_provider/tests/test_users.py

```diff
@@ -155,3 +172,71 @@ def test_unique_username_created(self, username_mock):
         self.assertEqual(User.objects.count(), 2)
         user = User.objects.get(username='new_edx_id')
         self.assertEqual(user.email, 'new_edx_id@lti.example.com')
+
+
+class LtiBackendTest(TestCase):
+    """
+    Tests for the authentication backend that authenticates LTI users.
+    """
+
+    def setUp(self):
+        super(LtiBackendTest, self).setUp()
+        self.edx_user = UserFactory.create()
+        self.edx_user.save()
+        self.lti_consumer = LtiConsumer(
+            consumer_key="Consumer Key",
+            consumer_secret="Consumer Secret"
+        )
+        self.lti_consumer.save()
+        self.lti_user_id = 'LTI User ID'
+        LtiUser(
+            lti_consumer=self.lti_consumer,
+            lti_user_id=self.lti_user_id,
+            edx_user=self.edx_user
+        ).save()
+
+    def test_valid_user_authenticates(self):
+        user = users.LtiBackend().authenticate(
+            username=self.edx_user.username,
+            lti_user_id=self.lti_user_id,
+            lti_consumer=self.lti_consumer
+        )
+        self.assertEqual(user, self.edx_user)
+
+    def test_missing_user_returns_none(self):
+        user = users.LtiBackend().authenticate(
+            username=self.edx_user.username,
+            lti_user_id='Invalid Username',
+            lti_consumer=self.lti_consumer
+        )
+        self.assertIsNone(user)
+
+    def test_non_lti_user_returns_none(self):
+        non_edx_user = UserFactory.create()
+        non_edx_user.save()
+        user = users.LtiBackend().authenticate(
+            username=non_edx_user.username,
+        )
+        self.assertIsNone(user)
+
+    def test_missing_lti_id_returns_null(self):
+        user = users.LtiBackend().authenticate(
+            username=self.edx_user.username,
+            lti_consumer=self.lti_consumer
+        )
+        self.assertIsNone(user)
+
+    def test_missing_lti_consumer_returns_null(self):
+        user = users.LtiBackend().authenticate(
+            username=self.edx_user.username,
+            lti_user_id=self.lti_user_id,
+        )
+        self.assertIsNone(user)
```

Comment:
I'm not sure that the tests would be that easy to convert; most of the parameters being passed to the authenticate method are class variables (which the ddt decorator doesn't see), and test_non_lti_user_returns_none creates a new User model. Unless I'm missing something simple, it seems like converting the tests to ddt would seem to make them less understandable.